h2o-core/src/main/java/hex/ParallelModelBuilder.java

```diff
@@ -18,13 +19,27 @@
 
   public static abstract class ParallelModelBuilderCallback<D extends ParallelModelBuilderCallback> extends Iced<D> {
 
-    public abstract void onBuildSucces(final Model model, final ParallelModelBuilder parallelModelBuilder);
+    public abstract void onBuildSuccess(final Model model, final ParallelModelBuilder parallelModelBuilder);
 
     public abstract void onBuildFailure(final ModelBuildFailure modelBuildFailure, final ParallelModelBuilder parallelModelBuilder);
+
+    public abstract void afterBuildProcessing(final ParallelModelBuilder parallelModelBuilder, final Model model);
   }
 
   private final transient ParallelModelBuilderCallback _callback;
-  private final transient IcedAtomicInt _modelInProgressCounter = new IcedAtomicInt();
+
+  public AtomicInteger getModelInProgressCounter() {
```

Comment:
this exposes the internal state unecessarily, this method and the one below can return simple int

Reply:
yes, great point

why are the counters not transient anymore?

Reply:
@honzasterba  Moving `hasReachedMaxModels` to ParallelModelBuilder would help to avoid exposure

Reply:
@michalkurka will add transient keywords. Btw... how do we make sure that our copy of ForkJoinTask is up to date?